djangocms_versioning/plugin_rendering.py

```diff
@@ -50,15 +50,12 @@ def render_obj_placeholder(
         ):
             # FIXME This is an ad-hoc solution for page-specific rendering
             # code, which by default doesn't work well with versioning.
-            # Remove this method once the issue is fixed.
-            from cms.models import Placeholder
 
             current_obj = self.toolbar.get_object()
 
             # Not page, therefore we will use toolbar object as
             # the current object and render the placeholder
-            rescan_placeholders_for_obj(current_obj)
-            placeholder = Placeholder.objects.get_for_obj(current_obj).get(slot=slot)
+            placeholder = rescan_placeholders_for_obj(current_obj).get(slot)
```

Comment:
Is it possible/easy to get a test for this?

Reply:
@marksweb Current tests cover that the correct placeholder is returned (I checked that by breaking it). So, I believe functionality is covered. This change just avoids repeating the same query again/querying the just created new placeholder, reducing the DB hits by 1.

I'm preparing a similar change and an accompanying test for the core which will cover to total number of queries for the edit and structure endpoints. 

I'd rather undo the change here than adding a compilcate query count here which largely depends on the django CMS core minor version.

What do you think?